Standards/scs-0111-v1-volume-type-decisions.md

```diff
@@ -0,0 +1,90 @@
+
+**Conclusion**: It might be noteable, that this already is a discoverable option.
+
+#### Replication
+
+Replication states, whether or not there are multiple replicas of a volume. Thus answers the question, whether the data could survive a node outage. Again there are different ways to achive replicated volumes. It can either be defined in the volume type and is discoverable also by normal users or it is configured in the backend. The last option is usually used with ceph for example. This makes it hard to discover, whether a volume is replicated or not. Another point is the number of replicas, that exist.
+
+**Conclusion**: Replication is a good option to be standardized. Whether this should be done as a boolean option or if the number of replicas is also something users need to know should still be discussed. Nevertheless due to the different options to configure replication this will be quite complex.
+
+#### QoS
+
+Quality of Service parameters can be stated in a volume qos object. These objects can then be associated to a volume type (or directly to a volume as an admin only option). But this is optional and thus even good or very good volume QoS parameters that are aquired through hardware configuration and storage parameters, might go by unmentioned.
+Furthermore the indirection makes it harder to discover the qos for a volume type. Only admins will see the associated qos ID and will have to take a closer look at the qos after discovering the volume type. PLUS: there can only be one qos association for one volume type. But a qos can be used for multiple volumes.
+
+**Conclusion**: The benefit of displaying qos parameters is clear, thus this option should be noted. But are volume qos objects widely used? If not, standardization process would be too much work.
+
+## Open questions
+
+1. How often are the different options used by providers and users respectively? Especially important for qos and replication!
+2. Regarding Replication: Is the number of replicas needed by users and is it okay for providers to provide this information?
+
+## Decision
+
+| Aspect | Standardize? | Discoverability | other Things |
+| ---- | ---- | ---- | ------ |
+| encryption | **Recommended** | work needed | extra_spec: encrypted=True/False |
+| Backend name | - | - | - |
+| AZs | - | - | describe as optional and backend-dependend |
+| multiattach | - | yes | describe as optional |
+| Replication | **Recommended** | lot of work | either get from backend to OS or as extra_spec defined by deployer |
+| Number of Replicas, etc | ? | lot of work | optional, work on it after Replication is standardized |
+| Volume QoS | ? | admin only | needs further discussion, should be at least described as optional |
+
```

Comment:
Should storage backend features (like snapshotting, CoW, ..) be visible?

Reply:
Should it be possible to have something for the purpose of the storage backend? Read-optimized, write-optmized, ...